code/modules/clothes_making/weaving.dm

```diff
@@ -0,0 +1,383 @@
+			to_chat(user, "<span class='warning'>There is no flax in the bag.</span>")
+	else
+		..()
+
+/obj/structure/spinning_wheel/process()
+	set waitfor = FALSE
+
+	if (!spinner || !Adjacent(spinner) || spinner.incapacitated() || spinner.lying || (remaining_cloth_to_spin <= 0))
+		spinner = null
+		update_icon()
+		processing_objects.Remove(src)
+		return
+	if (remaining_cloth_to_spin > 0)
+		playsound(src, 'sound/machines/loom_wooden.ogg', 20, 0)
+		if (spawn_cloth(spinner))
+			spawn(10)//process is called every 2 seconds, this lets us spawn 1 cloth per second
+				spawn_cloth(spinner)
+
+
+/obj/structure/spinning_wheel/proc/spawn_cloth(var/mob/_spinner)
+	remaining_cloth_to_spin--
+	drop_stack(/obj/item/stack/sheet/cloth, loc, 1)
+	if (remaining_cloth_to_spin <= 0)
+		to_chat(_spinner, "<span class='warning'>There [src] is out of flax.</span>")
+		spinner = null
+		update_icon()
+		processing_objects.Remove(src)
+		return 0
+	return 1
+
+/obj/structure/spinning_wheel/update_icon()
+	if (spinner)
+		icon_state = "wooden_loom_spin"
+	else if (remaining_cloth_to_spin > 0)
+		icon_state = "wooden_loom_ready"
+	else
+		icon_state = "wooden_loom"
+
+
+///////////////////ELECTRIC LOOM//////////////////////////////////////////////////////////////////////////////////////////
+
+/obj/machinery/electric_loom
+	name = "electric loom"
+	desc = "Automatically turns flax into cloth while powered. You can set input and output directions with a multitool."
+	icon = 'icons/obj/clothes_making.dmi'
+	icon_state = "electric_loom"
+	density = 1
+	anchored = 1
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | FIXED2WORK | EJECTNOTDEL | MULTIOUTPUT
+	use_power = MACHINE_POWER_USE_IDLE
+	idle_power_usage = 5
+	active_power_usage = 500
+	pass_flags_self = PASSMACHINE
+
+	var/remaining_cloth_to_spin = 0
+	var/current_production = 0
+	var/stored_cloth = 0
+
+	var/manipulator_rating = 0
+	var/matterbin_rating = 0
+
+/obj/machinery/electric_loom/New()
+	. = ..()
+
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/electric_loom,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/matter_bin,
+		/obj/item/weapon/stock_parts/matter_bin,
+	)
+
+	RefreshParts()
+	update_icon()
+
+/obj/machinery/electric_loom/examine(mob/user)
+	..()
+	if(remaining_cloth_to_spin + current_production > 0)
+		to_chat(user, "<span class='info'>There is enough flax in it to produce [remaining_cloth_to_spin*matterbin_rating + current_production] more cloth sheets.</span>")
+	else
+		to_chat(user, "<span class='info'>Grow some flax then put it in there before you can use it. You can insert flax in it automatically using a conveyor belt.</span>")
+
+	if(stored_cloth > 0)
+		to_chat(user, "<span class='info'>There's [stored_cloth] lengths of cloth on the roll currently being spun. The machine will automatically eject it when its full, or you can pick it up now.</span>")
```

Comment:
There are
eject them
it's
pick them

of course all of these imply plurality which it might not be, you could rephrase to work with any plurality with:
`There are [stored_cloth] lengths of cloth on the roll being spun. The machine will eject when it is full, or you can eject the roll now.`